### PR TITLE
Make oximeter producer kind required

### PR DESCRIPTION
**Pull request by bnaecker**
- Pulls in updated Dendrite, Propolis, and Crucible deps, which include the new producer kind enum in metric registration requests. From their perspective, this is still an optional parameter, but it is supplied.
- Make the kind a required field in API requests.
- Make the kind a required column in the database, and remove any rows with a NULL value.
- Update OpenAPI documents and internal consumers to reflect the required parameter.

### CODE REVIEW

**Comment by bnaecker** [comment]
Just a note, it looks like #4559 is also targeting a schema version of 15, so I'll change this one to 16 if that merges first.

**Review by leftwo — COMMENTED** [review]
If we bump Crucible rev here, but use a propolis that has the previous
rev, do things work as expected?  I think this may be the first time we tried
this after the work Patrick did to support it.

File: package-manifest.toml [hunk 1]
```diff
@@ -384,21 +384,21 @@ only_for_targets.image = "standard"
 # 3. Use source.type = "manual" instead of "prebuilt"
 source.type = "prebuilt"
 source.repo = "crucible"
-source.commit = "51a3121c8318fc7ac97d74f917ce1d37962e785f"
+source.commit = "945f040d259ca8013d3fb26f510453da7cd7b1a6"
 # The SHA256 digest is automatically posted to:
-# https://buildomat.eng.oxide.computer/public/file/oxidecomputer/crucible/image/<commit>/crucible.sha256.txt
-source.sha256 = "897d0fd6c0b82db42256a63a13c228152e1117434afa2681f649b291e3c6f46d"
+# https://buildomat.eng.oxide.computer/public/file/oxidecomputer/crucible/rbuild/<commit>/crucible.sha256.txt
```
[the remark this answers is not in the document]
Comment (bnaecker) [reply package-manifest.toml:389]: Oh, thanks for catching that. I conflated the buildomat public URL with the GH checksuite name. I'll replace it.

**Comment by bnaecker** [comment]
> If we bump Crucible rev here, but use a propolis that has the previous rev, do things work as expected? 

I'm not sure, but I think it depends on how old the Propolis revision is. If it's prior to the addition of the optional parameter, then the metric registration request will fail. After that, things should be OK.

**Comment by leftwo** [comment]
> > > If we bump Crucible rev here, but use a propolis that has the previous rev, do things work as expected?
> > 
> > 
> > I'm not sure, but I think it depends on how old the Propolis revision is. If it's prior to the addition of the optional parameter, then the metric registration request will fail. After that, things should be OK.
> 
> I looks like current propolis that Omicron is using is: 54398875a2125227d13827d4236dce943c019b1c
> 
> That is before your `Add kind field to metric registration request (#568)` PR. However. I took your bits and put them on my bench gimlet, and I believe things are working. I can't SSH into my instance, but I'm able to stop/start instances, see output on the console, and see metrics change value after I boot/reboot my instance, so it looks like my concern is unwarrented.

Oh, wait, the propolis version here is also updated, so my concern is not warranted.
I don't know why I was thinking something different here.

**Review by leftwo — APPROVED** [review]

**Comment by bnaecker** [comment]
> Oh, wait, the propolis version here is also updated, so my concern is not warranted.

Ok this explains the confusion! For the history books: this PR updates _all of_ Crucible, Propolis, and Dendrite to versions which provide the producer kind parameter in metric registration requests. Those out-of-tree consumers think the value is _optional_, because it is optional in the OpenAPI documents in the Omicron `main` against which they're built. But because an optional and required value look the same on the wire, both sides should be happy. In other words, we're at step 5 @davepacheco's list above.

**Pull request merged**